### ISSUE TITLE
Dev-guide

### ISSUE THREAD

**Issue by scrabsha** [iomentum/cargo-breaking#11]
It would be a nice idea to have some documentation that describe briefly how things work. This would make contributions easier for newcomers.

**Comment by cbonaudo** [comment]
Are you speaking about each mod specifically (eg: doc for git.rs, doc for glue.rs, etc...) ?

**Comment by scrabsha** [comment]
Yep! where each thing happen, how data flows between each pass, and so on.
Maybe we can take inspiration from [Overview of the Compiler][ootc] from the rustc dev guide.

[ootc]: https://rustc-dev-guide.rust-lang.org/overview.html

**Comment by cbonaudo** [comment]
Created in #15, with a brief summary.
now we can expand each part to describe accurately what is done
